thrift/fields.go

```diff
@@ -103,6 +121,25 @@ func fieldsMapToValue(fields map[string]*compile.FieldSpec, userFields map[strin
 	return wireFields, nil
 }
 
+const upperToLower = 'a' - 'A'
+
+// fuzz returns a copy of fieldName that is suitable for fuzzy matching.
+// It lowercases the string, and removes non-alphanumeric characters.
+func fuzz(fieldName string) string {
+	newField := make([]byte, 0, len(fieldName))
+	for i := 0; i < len(fieldName); i++ {
+		c := fieldName[i]
+		switch {
+		case c >= 'A' && c <= 'Z':
+			newField = append(newField, c+upperToLower)
+		case c >= 'a' && c <= 'z', c >= '0' && c <= '9':
+			newField = append(newField, c)
```

Comment:
Couple of reasons:
 - Since I'm only dealing with a very small set of known characters (e.g. based on the Thrift spec), we only really care about ASCII.
 - Unicode deals with runes which can't be easily appended to a `[]byte` without converting to a string. In general, there's more overhead